src/components/ReviewForm.js

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import axios from "axios";
+
+export function ReviewForm(props) {
+  const { id } = props.editableReview;
+  const [errorTitleMessage, setErrorTitleMessage] = useState("");
+  const [errorScoreMessage, setErrorScoreMessage] = useState("");
+  const [errorBodyMessage, setErrorBodyMessage] = useState("");
+  const [errorReviewerMessage, setErrorReviewerMessage] = useState("");
+  const [formData, setFormData] = useState(
+    id
+      ? props.editableReview
+      : {
+          title: "",
+          score: 1,
+          body: "",
+          reviewer: "",
+        }
+  );
+  const actionText = id ? "レビューを編集する" : "レビューを登録する";
+
+  function handleChange({ target }) {
+    const value = target.value;
+    console.log(value);
+    setFormData({
+      ...formData,
+      [target.name]: value,
+    });
+  }
+
+  // async関数にしてあげる　updateReviewをawaitするために
+  async function handleSubmit(event) {
+    event.preventDefault(); // 画面遷移してしまわないように必要
+    console.log(formData);
+    if (validate(formData)) {
+      if (id) {
+        await updateReview(id);
+      } else {
+        postReview();
+      }
+    }
+    props.closeModal();
+  }
+
+  function validate({ title, score, body, reviewer }) {
+    if (title && title.length <= 255) {
+      setErrorTitleMessage("");
+    } else {
+      setErrorTitleMessage("Title is invalid");
+    }
+
+    if (score && 0 < score <= 5) {
```

Comment:
気持ちはわかりますがこの書き方は期待通りに動かないので分けて書きましょう

```suggestion
    if (score && 0 < score && score <= 5) {
```

ちなみに

```js
> let score = -1
> 0 < score <= 5
true
```